student_explorer/urls.py

```diff
@@ -30,8 +30,11 @@
 ]
 
 if 'djangosaml2' in settings.INSTALLED_APPS:
+    from djangosaml2.views import echo_attributes
+    from django.contrib.auth.decorators import login_required
     urlpatterns += (
         url(r'^accounts/', include('djangosaml2.urls')),
+        url(r'^accounts/echo_attributes/', login_required(echo_attributes)),
```

Comment:
The problem with login_required is it won't work if you get permission denied. . . 

Maybe this login_required doesn't need to be checked. . . Going to check.

Reply:
Maybe on a failed login we can have it log the attributes that failed . . .